app/models/miq_expression/field.rb

```diff
@@ -29,11 +30,26 @@ def datetime?
     column_type == :datetime
   end
 
+  def to_s
+    assoc = associations.map { |a| ".#{a}" }.join
+    "#{model.name}#{assoc}-#{column}"
+  end
+
+  def reflections
+    klass = model
+    associations.collect do |association|
+      klass.reflect_on_association(association).tap do |reflection|
+        reflection or raise InvalidFieldError, self
```

Comment:
How will error reporting work?

I know the code I work with wants to say "field X is not valid, rather than list everything"
a little tricky